lib/gems/pending/util/mount/miq_s3_session.rb

```diff
@@ -0,0 +1,66 @@
+require 'util/mount/miq_generic_mount_session'
+require 'aws-sdk'
+
+class MiqS3Session < MiqGenericMountSession
+  def initialize(log_settings)
+    log_header = "MIQ(#{self.class.name}-initialize)"
+    logger.debug("#{log_header} initialize: log_settings are #{log_settings}")
+    super(log_settings)
+    logger.debug("#{log_header} initialize: @settings are #{@settings}")
+    raise "username is a required value!" if @settings[:username].nil?
+    raise "password is a required value!" if @settings[:password].nil?
+    raise "region is a required value!" if @settings[:region].nil?
+    @s3 = s3
```

Comment:
I don't understand this line.  `@s3` is set in the s3 method already, so why set it again?  On top of that `@s3` is memoized on first access, so why do that if you are going to just set it during initialize anyway.